src/coreclr/src/jit/gentree.cpp

```diff
@@ -12486,32 +12487,19 @@ GenTree* Compiler::gtFoldTypeCompare(GenTree* tree)
         return tree;
     }
 
-    // We must have a handle on one side or the other here to optimize,
-    // otherwise we can't be sure that optimizing is sound.
-    const bool op1IsFromHandle = (op1Kind == TPK_Handle);
-    const bool op2IsFromHandle = (op2Kind == TPK_Handle);
-
-    if (!(op1IsFromHandle || op2IsFromHandle))
-    {
-        return tree;
-    }
-
     // If both types are created via handles, we can simply compare
-    // handles (or the indirection cells for handles) instead of the
-    // types that they'd create.
-    if (op1IsFromHandle && op2IsFromHandle)
+    // handles instead of the types that they'd create.
```

Comment:
I do not think there is any interesting case where the optimization for indirection cells kick in. It looks like a left-over from the past.